pokt-network/pocket
# [Telemetry] Add event metrics for RainTree

## Description


This PR adds event-metrics through to Pre2P RainTree implementation using the Telemetry module.

## Changes
Please mark the options that are relevant.

- [x] Added instrumentation to record the reception of a RainTree message as an event for each block, this event has the following attributes:
     - namespace `pre2p`
     - event_name `raintree_messages`
     - labels: [height]

- [x] Added instrumentation to record the reception of redundant RainTree messages as an event for each new block, the event has the following attributes:
     - namespace `pre2p`
     - event_name `broadcast_message_redundancy_per_block`
     - labels: [height, message_hash] 

- [x] Added `logql` query to process the recorded events and fed the results to a numerical series graph

## How Has This Been Tested?
__

1. Spin up the stack
```zsh
$ make compose_and_watch
```

2. Make sure you spin the client up, and that kick the blockchain off:
```zsh
$ make client_start && make client_connect
```

3. Choose:
```
✔ TogglePacemakerMode
✔ TriggerNextView
```

4. Open Grafana at [http://localhost:3000](http://localhost:3000)
5. Open the `RainTree Telemetry` graph

6. Verify the results on the dashboard

### Additional items for review

The query that was used to generate the graph is the following:

1. Total # RainTree Msgs / Block
```sql
sum by (blockheight) (
   count_over_time({host="docker-desktop"}
   |= "[EVENT] consensus"
   | pattern `<datetime> <_> <time> <type> <_> <metric_name> <_> <blockheight> <step> <validator_role>`[6s]
   )
),
```

2. Avg # Raintree Msgs / Block

```sql
avg by (blockheight, instance) (
  sum by (blockheight) (
    count_over_time({host="docker-desktop"}
    |= "[EVENT] p2p raintree_messages"
    | pattern `<datetime> <_> <time> <type> <namespace> <metric> <_> <blockheight>`[6s])
  )
)
```

3. Avg. Redundancy / Block
```sql
avg by (blockheight, instance) (
  sum by (blockheight) (
    count_over_time({host="docker-desktop"}
    |= "[EVENT] p2p broadcast_message_redundancy_per_block"
    | pattern `<datetime> <_> <time> <type> <_> <redundancy> <_> <hashstring> <_> <blockheight>`[6s])
   )
)
```

Owners: @derrandz 
Co-owners: @okdas @Olshansk
